File: ming/0915/Best Time to Buy and Sell Stock.java
public class Solution {
    public int maxProfit(int[] prices) {
        int len = prices.length;
        if (len < 2){
            return 0;
        }
        int min = prices[0];
        int maxProf = Integer.MIN_VALUE;
        int[] profit = new int[len];
        
        for (int i = 1; i < len; i++){
            min = Math.min(min, prices[i]);
            profit[i] = prices[i] - min;
            maxProf = Math.max(maxProf, profit[i]);
        }
        return maxProf;
    }
}

File: mu_wang/9_15/Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices)<1:
            return 0
        min_p = prices[0]
        profit = 0
        for i in range(0,len(prices)):
            min_p = min(min_p,prices[i])
            cha = prices[i]-min_p
            profit = max(cha,profit)
        
        return profit

a = Solution()
print a.maxProfit([1,2,1,5,4])